test/labeling/model/test_label_model.py

```diff
@@ -247,5 +250,63 @@ def test_train_model(self):
         self.assertLessEqual(next_loss, init_loss)
 
 
+@pytest.mark.complex
+class LabelModelTestAdvanced(unittest.TestCase):
+    """Advanced (marked complex) tests for the LabelModel."""
+
+    def setUp(self) -> None:
+        """Set constants for the tests."""
+        self.m = 10  # Number of LFs
+        self.n = 10000  # Number of data points
+        self.k = 2  # Number of classes
+
+    def _generate_data(self) -> Tuple[np.ndarray, np.ndarray, csr_matrix]:
```

Comment:
I'm a little torn on this. A bit complicated to put in `test` and we want to make sure it works. What's the current usefulness of `snorkel/synthetics`? Should we burn that and replace it with simpler stuff like this>

Reply:
Sure makes sense- different generators for different things, but I'll at least split this (and the corresponding test) into `snorkel/synthetics`